pkg/cmd/taskrun/logs_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taskrun
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/jonboulle/clockwork"
+	"github.com/knative/pkg/apis"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/logs"
+	tu "github.com/tektoncd/cli/pkg/test"
+	cb "github.com/tektoncd/cli/pkg/test/builder"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/test"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func Test_command_has_taskrun_arg(t *testing.T) {
+	c := Command(&tu.Params{})
+
+	_, err := tu.ExecuteCommand(c, "logs", "-n", "ns")
+
+	if err == nil {
+		t.Error("Expecting an error but it's empty")
+	}
+}
+
+func Test_taskrun_not_found(t *testing.T) {
+	tr := []*v1alpha1.TaskRun{
+		tb.TaskRun("output-task-1", "ns",
+			tb.TaskRunStatus(
+				tb.PodName("output-task-pod-123456"),
+				tb.Condition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+	}
+	cs, _ := test.SeedTestData(test.Data{TaskRuns: tr})
+	p := &tu.Params{Tekton: cs.Pipeline, Kube: cs.Kube}
+
+	c := Command(p)
+	got, _ := tu.ExecuteCommand(c, "logs", "output-pipeline-2", "-n", "ns")
```

Comment:
output-pipeline-2? copy paste error? 
also, how about
 
```
tr := "output-task-missing"
got, _ := tu.ExecuteCommand(c, "logs", tr, "-n", "ns")

expected := fmt.Sprintf(`%s : taskruns.tekton.dev "%s" not found%s`, msgTRNotFound, tr, "\n")
```

Reply:
Actually, we are setting fake taskrun with output-pipeline-1 and running the command to get logs of output-pipeline-2

Reply:
what I meant was, why are we calling it a `pipeline` ? isn't this a task -> taskrun ?